src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/Razor/GenerateMethodCodeActionResolver.cs

```diff
@@ -0,0 +1,112 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.AspNetCore.Razor.LanguageServer.CodeActions.Models;
+using Microsoft.AspNetCore.Razor.LanguageServer.Common;
+using Microsoft.AspNetCore.Razor.LanguageServer.Formatting;
+using Microsoft.AspNetCore.Razor.Utilities;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Razor;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Newtonsoft.Json.Linq;
+using CSharpSyntaxFactory = Microsoft.CodeAnalysis.CSharp.SyntaxFactory;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.CodeActions.Razor;
+internal class GenerateMethodCodeActionResolver : IRazorCodeActionResolver
+{
+    public string Action => LanguageServerConstants.CodeActions.GenerateMethod;
+    private readonly DocumentContextFactory _documentContextFactory;
+    private readonly RazorLSPOptions _razorLSPOptions;
+    private static readonly string s_generatedCode = $$"""0private void $$MethodName$$(){{Environment.NewLine}}0{{{Environment.NewLine}}1throw new System.NotImplementedException();{{Environment.NewLine}}0}""";
```

Comment:
I think we would still end up needing a double `Replace` call. `$$Indent$$` includes if there is any initial indent. e.g. if it were an initial indent of 1 (for whatever reason):
```
   @code {
   }
```
instead of no initial indent: 
```
@code {
}
```

I realize that that was not clear so I've changed it to `$$InitialIndent$$` and `$$Indent$$` so hopefully it is more clear.
Upon doing that I also realized that the initial indent that is read from what is on the file can either be `\t` or ` ` so I ended up needing to add some extra logic to `FormattingUtilities` to account that `\t` = `TabSize` amount of characters.

Reply:
ahh, yes, you have correctly predicted that I did not understand the difference between the two indents. Much clearer :D